### DISCUSSION TITLE
How to use the bearer token from a variable in Illa's restful API data source feature?

### DISCUSSION THREAD

**Discussion by jamin-xie** [illacloud/illa-builder#3647] [English Support]
My API's bearer token is not fixed, and it has an expiration time. 
Such a fixed token cannot meet my needs. Therefore, can Illa's restful API set a variable or dynamic parameter as the token?

**Comment by AruSeito** [comment]
If you have an API that can retrieve tokens, you can try doing the following:

1. Create a new Action to get a Token and name it Action1

2. Set the token to {{action1.data.xxxxxx}} within the existing Action to replace the previous fixed token
